### DISCUSSION TITLE
Returning empty list when selecting "IS NULL"

### DISCUSSION THREAD

**Discussion by AndAccioly** [supabase/supabase#32423] [Questions]
I have a table with some rows containing null values on one column. I am trying to select these rows, but supabase always return with empty list, even on SQL editor:

SQL editor: select * from imoveis where valor_venda IS NULL
react: supabase.from('imoveis').select('*').is('valor_venda', null)

I don't know if I forgot any configuration, but it won't workd even if I have hundreds of null data. The column is a float4 type. Does anyone know how to fix it?

**Comment by GaryAustin1** [comment]
That should work and does:
![image](https://github.com/user-attachments/assets/0d976139-5fea-41b9-a9c2-0913602bbf0a)

Are you sure you don't have RSL enabled and policy not met?  This would also mean you are running impersonation mode in the SQL editor though.

**Comment by AndAccioly** [comment]
It is strange because it works for pretty any select, just the is null does not return data. Example:

![image](https://github.com/user-attachments/assets/21cffeb7-85f0-40aa-81b9-95d9580e195f)

I have disabled RSL for all my tables, and I can show here that there are null values:

![image](https://github.com/user-attachments/assets/66a0e8de-8f3e-414f-84da-0c05fc74fc9e)

What I am doing is making 0 as default for this column so I won't have more nulls, it is a workaround but does not fix the problem.

**Reply by GaryAustin1** [reply]
Try it on another table.  Not sure what to say as using the same column type and SQL query it works for me.
